RecoLocalCalo/HcalRecProducers/python/HBHEPhase1Reconstructor_cfi.py

```diff
@@ -0,0 +1,86 @@
+    # whether we should process this collection
+    digiLabelHE = cms.InputTag("hcalDigis"),
+    processHE = cms.bool(True),
+
+    # Get the "sample of interest" index from DB?
+    # If not, it is taken from the dataframe.
+    tsFromDB = cms.bool(False),
+
+    # Use the HcalRecoParam structure from DB inside
+    # the reconstruction algorithm?
+    recoParamsFromDB = cms.bool(True),
+
+    # Drop zero-suppressed channels?
+    dropZSmarkedPassed = cms.bool(True),
+
+    # Flag indicating whether we should produce HBHERecHitCollection
+    makeRecHits = cms.bool(True),
+
+    # Flag indicating whether we should produce HBHEChannelInfoCollection
+    saveInfos = cms.bool(False),
+
+    # Flag indicating whether we should include HBHEChannelInfo objects
+    # into HBHEChannelInfoCollection despite the fact that the channels
+    # are either tagged bad in DB of zero-suppressed. Note that the rechit
+    # collection will not include such channels even if this flag is set.
+    saveDroppedInfos = cms.bool(False),
+
+    # Configure the reconstruction algorithm
+    algorithm = cms.PSet(
+        # Parameters for "Method 3" (non-keyword arguments have to go first)
+        method3.m3Parameters,
+
+        Class = cms.string("SimpleHBHEPhase1Algo"),
+
+        # Time shift (in ns) to add to TDC timing (for QIE11)
+        tdcTimeShift = cms.double(0.0),
+
+        # Parameters for "Method 0"
+        firstSampleShift  = cms.int32(0),
+        samplesToAdd      = cms.int32(2),
+        correctionPhaseNS = cms.double(6.0),
+
+        # Use "Method 2"? Change this to True when implemented.
+        useM2 = cms.bool(False),
+
+        # Parameters for "Method 2"
+        applyPedConstraint    = cms.bool(True),
+        applyTimeConstraint   = cms.bool(True),
+        applyPulseJitter      = cms.bool(False),  
+        applyUnconstrainedFit = cms.bool(False),  #Turn on original Method 2
+        applyTimeSlew         = cms.bool(True),   #units
+        ts4Min                = cms.double(0.),   #fC
+        ts4Max                = cms.double(100.), #fC
+        pulseJitter           = cms.double(1.),   #GeV/bin
+        meanTime              = cms.double(0.),   #ns
+        timeSigma             = cms.double(5.),   #ns
+        meanPed               = cms.double(0.),   #GeV
+        pedSigma              = cms.double(0.5),  #GeV
+        noise                 = cms.double(1),    #fC
+        timeMin               = cms.double(-12.5),#ns
+        timeMax               = cms.double(12.5), #ns
+        ts3chi2               = cms.double(5.),   #chi2 (not used)
+        ts4chi2               = cms.double(15.),  #chi2 for triple pulse 
+        ts345chi2             = cms.double(100.), #chi2 (not used)
+        chargeMax             = cms.double(6.),   #Charge cut (fC) for uncstrianed Fit 
+        fitTimes              = cms.int32(1),     # -1 means no constraint on number of fits per channel
```

Comment:
No, I presume that there will be a different set of parameters for
HB and HE. However, the exact breakdown of that is unknown
at the moment (which parameters will be common and which will
be different? etc). So, it does not make much sense to refactor
the python definitions at the moment. We'll clean up the dust when
the construction is complete.

On 07/01/2016 08:51 AM, Slava Krutelyov wrote:
>
> In RecoLocalCalo/HcalRecProducers/python/HBHEPhase1Reconstructor_cfi.py <https://github.com/cms-sw/cmssw/pull/15092#discussion_r69302109>:
>
> > +        applyTimeSlew         = cms.bool(True),   #units
> > +        ts4Min                = cms.double(0.),   #fC
> > +        ts4Max                = cms.double(100.), #fC
> > +        pulseJitter           = cms.double(1.),   #GeV/bin
> > +        meanTime              = cms.double(0.),   #ns
> > +        timeSigma             = cms.double(5.),   #ns
> > +        meanPed               = cms.double(0.),   #GeV
> > +        pedSigma              = cms.double(0.5),  #GeV
> > +        noise                 = cms.double(1),    #fC
> > +        timeMin               = cms.double(-12.5),#ns
> > +        timeMax               = cms.double(12.5), #ns
> > +        ts3chi2               = cms.double(5.),   #chi2 (not used)
> > +        ts4chi2               = cms.double(15.),  #chi2 for triple pulse
> > +        ts345chi2             = cms.double(100.), #chi2 (not used)
> > +        chargeMax             = cms.double(6.),   #Charge cut (fC) for uncstrianed Fit
> > +        fitTimes              = cms.int32(1),     # -1 means no constraint on number of fits per channel
>
> HB will stay with legacy readout for a while.
> If you are saying that the M2 parameters will need to be adjusted for SiPM for HE, then I don't understand how HB and HE will be treated simultaneously with this module PSet.
> I can imagine there is identical M2 config for this module and the HcalHitReconstructor for HB and then HE 2017 will have its own settings.
>
> please elaborate.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/cms-sw/cmssw/pull/15092/files/6d2850b92942e3626e740059aff5348d603d1242#r69302109>, or mute the thread 
> <https://github.com/notifications/unsubscribe/AFS4-QTmknsICj_G5YI35_geqvNDtEeZks5qRRtcgaJpZM4JCc97>.
>